appserver/concurrent/concurrent-impl/src/main/java/org/glassfish/concurrent/runtime/ContextSetupProviderImpl.java

```diff
@@ -131,9 +131,7 @@ public ContextHandle setup(ContextHandle contextHandle) {
         ComponentInvocation invocation = invocationCtx.getInvocation();
         final String appName = invocation == null ? null : invocation.getAppName();
 
-        if (!isApplicationEnabled(appName)) {
-            throw new IllegalStateException("Module " + appName + " is disabled");
-        }
+        verifyApplicationEnabled(appName);
```

Comment:
Hmm, you check for null here, on original invocation, before setting up current invocation. Or it setting up somewhere else?

Reply:
Yes. My assumption is that the original invocation is always set and contains the appName. The current invocation never contains the app name (current invocation is actually null all the time, unless the app modifies it with some GlassFish API if that’s possible). This method sets the appName of the current invocation if it’s null.